posthog/models/plugin.py

```diff
@@ -70,3 +166,37 @@ class Meta:
     plugin_config: models.ForeignKey = models.ForeignKey("PluginConfig", on_delete=models.CASCADE)
     key: models.CharField = models.CharField(max_length=200)
     value: models.TextField = models.TextField(blank=True, null=True)
+
+
+@receiver(models.signals.post_save, sender=Organization)
+def preinstall_plugins_for_new_organization(sender, instance: Organization, created: bool, **kwargs):
+    if created and not settings.MULTI_TENANCY and not settings.TEST and can_install_plugins(instance):
+        # This in disabled in tests to avoid hitting GitHub API limits
+        for plugin_url in settings.PLUGINS_PREINSTALLED_URLS:
+            Plugin.objects.install(
+                organization=instance, plugin_type=Plugin.PluginType.REPOSITORY, url=plugin_url, is_preinstalled=True
+            )
+
+
+@receiver(models.signals.post_save, sender=Team)
+def enable_preinstalled_plugins_for_new_team(sender, instance: Team, created: bool, **kwargs):
+    if created and can_configure_plugins(instance.organization):
+        for order, preinstalled_plugin in enumerate(Plugin.objects.filter(is_preinstalled=True)):
+            PluginConfig.objects.create(team=instance, plugin=preinstalled_plugin, enabled=True, order=order)
```

Comment:
I think when configuring plugins via the UI, we send via POST a "config" json field with at least the default values. I don't know if the absence of those values makes things wonky or not, but I would be cautious here. Plugin authors might rely on defaults being there... and we just pass `pluginConfig.config` to the vm.

Probably best to inject the defaults in the plugin server if the keys are missing in the config object. So nothing to do here...

Reply:
Thanks for pointing this out, may be a problem in the future. For now it's not a problem as only the GeoIP plugin is preinstalled and it doesn't have config, but fixed that anyway for the future. Added `Plugin.get_default_config()` – for now only use here, but likely good to switch to that overall so that frontend doesn't need to send data extracted out of `Plugin.config_schema`?